### ISSUE TITLE
typescript file extension in tsconfig json path is not inferred

### ISSUE THREAD

**Issue by RobinMeow** [softarc-consulting/sheriff#49]
# .ts file extension in tsconfig json path is not inferred

## Reproduce

1. standard minimal project, no sheriff config needed, but install and add the "files" and "extend" properties in the `.eslintrc.json`
2. in your `tsconfig.json`, in the the `path` section, add a path mapping like this one:
```json
"paths": {
  "@shared/assertions": [ "src/domains/shared/assertions/index" ],
}
```
3. create a folder with an index.ts file (representing your module)

4. run your lint command (in my case, `yarn lint` / `ng lint`)

this produced the following error messages (and in more other files, but for simplicity I only show one file):

> C:\repos\CommunityCookbook\client\src\main.ts
>     1:1  error  Deep Import (internal error): FsPath: C:\repos\CommunityCookbook\client\src\domains\shared\assertions\index does not exist      @softarc/sheriff/deep-import
>     1:1  error  Dependency Rule (internal error): FsPath: C:\repos\CommunityCookbook\client\src\domains\shared\assertions\index does not exist  @softarc/sheriff/dependency-rule

![image](https://github.com/softarc-consulting/sheriff/assets/111490292/0c95dcaf-f270-47fd-943a-fbec64186156)

## Problem
The error is thrown in alot of files (I think in all .ts files), also note, that the `main.ts` looks like this: 
```typescript
import { AppComponent } from './app/app.component';
import { bootstrapApplication } from '@angular/platform-browser';
import { appConfig } from './app/app.config';

bootstrapApplication(AppComponent, appConfig).catch((err) =>
  console.error(err),
);
```
-> Meaning, it actually has no dependency to trigger the error.

## Workaround

simply add the file extension: 
from `"@shared/assertions": [ "src/domains/shared/assertions/index" ]`  
to -> `"@shared/assertions": [ "src/domains/shared/assertions/index.ts" ]`

## Extra Note

- The errors are thrown only for the first path-mapping. if you remove the first one, and re-run the command it will throw error for the next one.
- The error message where produced in this project: https://github.com/RobinMeow/CommunityCookbook/pull/59  
- pipeline with the errors: https://github.com/RobinMeow/CommunityCookbook/actions/runs/7179722928/job/19550507912?pr=59  
- the workaround commit: https://github.com/RobinMeow/CommunityCookbook/pull/59/commits/900fd43bec8e55feafbf3a8b1fcca53049210466
- the pipeline after the workaround commit with the desired sheriff-lint error messages: https://github.com/RobinMeow/CommunityCookbook/actions/runs/7180536003/job/19553019416

**Comment by rainerhahnekamp** [comment]
Hey, thanks for letting me know. That is indeed a bug in the paths resolution. Do you want to fix it or should I give it a try?

**Comment by RobinMeow** [comment]
Not sure if im experienced enough to do so, but I'd like to give it a try.

I'll put another comment in here, if to let you know, if I stopped trying (or a PR)

**Comment by RobinMeow** [comment]
_Originally posted by @rainerhahnekamp in https://github.com/softarc-consulting/sheriff/issues/46#issuecomment-1851895092_

I cant really get the pre-existing tests to pass. 

1. I cloned the repo `gh repo clone ...`
2. installed the packages using `yarn install --frozen-lockfile` to avoid updating packages 
3. ran `yarn test`  
  - ![image](https://github.com/softarc-consulting/sheriff/assets/111490292/a46a6727-092b-4628-b7af-7671b6b8aaab)
  - image is made after I installed jest `yarn add --dev jest`, because when I clicked run on a single test, it said "module not found" with the path to node_module/jest
4. after installing jest, the test still fails  
![image](https://github.com/softarc-consulting/sheriff/assets/111490292/83339967-5ff5-4a25-befa-7b531a43afde)

**Comment by rainerhahnekamp** [comment]
Oh sorry Robin, I see you are on Windows :(.

Yeah, so the tests depend on a UNIX-based filesystem notation. So, the directory separator is hard-coded as `/` instead of `\`. The reason for that was that all tests are based on a virtual filesystem. There would be too many files to create for the amount of tests we have.

Then let me do the fix and you try out the new version and tell me if it is working or not.

Apologies again.

**Comment by RobinMeow** [comment]
no worries, im just glad, it wasnt a layer 8 problem :D

**Comment by rainerhahnekamp** [comment]
This has been fixed in #53 and is available in version 0.14.3. @RobinMeow could you please check on your side and let me know if you encounter any issues?
